src/main/java/mate/academy/AsyncRequestProcessor.java

```diff
@@ -1,16 +1,38 @@
 package mate.academy;
 
+import java.util.Map;
 import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.Executor;
 
 public class AsyncRequestProcessor {
     private final Executor executor;
+    private Map<String, UserData> cache = new ConcurrentHashMap<>();
 
     public AsyncRequestProcessor(Executor executor) {
         this.executor = executor;
     }
 
     public CompletableFuture<UserData> processRequest(String userId) {
-        return null;
+        return CompletableFuture.supplyAsync(() -> getUserData(userId), executor);
+    }
+
+    private UserData getUserData(String userId) {
+        return cache.containsKey(userId) ? cache.get(userId) : collectUserData(userId);
+    }
+
+    private UserData collectUserData(String userId) {
```

Comment:
It's better to divide a logic of getting data with delay and adding the data to the cache to separate CompletableFuture callbacks, to make it truly asynchronous. Something like: `supplyAsync(/*get data*/).whenComplete(/*put data to the cache*/)`